File: Hackathon.Common/Models/Event/EventStatus.cs
namespace Hackathon.Common.Models.Event;

/// <summary>
/// Статус мероприятия
/// </summary>
public enum EventStatus
{
    /// <summary>
    /// Черновик
    /// </summary>
    /// <remarks>Могут вноситься любые правки</remarks>
    Draft = default,

    /// <summary>
    /// Опубликовано
    /// </summary>
    /// <remarks>
    /// Нельзя вносить изменения в общие параметры.
    /// Регистрация участников, команд
    /// </remarks>
    Published = 1,

    /// <summary>
    /// Событие начато
    /// </summary>
    /// <remarks>Регистрация закрыта. Организатор приветствует участников</remarks>
    Started = 2,

    /// <summary>
    /// Событие завершено
    /// </summary>
    /// <remarks>Все доступные ранее операции запрещены</remarks>
    Finished = 8
}

File: Hackathon.UI/src/app/models/EventStatus.ts
export enum EventStatus
{
  Draft = 0,
  Published = 1,
  Started = 2,
  Development = 3,
  Prepare = 4,
  Presentation = 5,
  Decision = 6,
  Award = 7,
  Finished = 8,

}

export enum TranslatedEventStatus
{
  Черновик = 0,
  Опубликовано = 1,
  Началось = 2,
  Разработка = 3,
  Подготовка = 4,
  Презентация = 5,
  'Выбор победителя' = 6,
  Награждение = 7,
  Закончено = 8,
}

export class Status{

  public static getTranslatedStatus = (e: EventStatus | undefined) : TranslatedEventStatus => {
    if ( e == 0) {
      return  TranslatedEventStatus.Черновик;}
    if ( e == 1) {
      return  TranslatedEventStatus.Опубликовано;}
    if ( e == 2) {
      return  TranslatedEventStatus.Началось;}
    if ( e == 3) {
      return  TranslatedEventStatus.Разработка;}
    if ( e == 4) {
      return  TranslatedEventStatus.Подготовка;}
    if ( e == 5) {
      return  TranslatedEventStatus.Презентация;}
    if ( e == 6) {
      return  TranslatedEventStatus["Выбор победителя"];}
    if ( e == 7) {
      return  TranslatedEventStatus.Награждение;}
    else
      return  TranslatedEventStatus.Закончено;
  }

  public static getTranslatedStatusValue(status:TranslatedEventStatus){
    return TranslatedEventStatus[status];
  }

}




